apm-sniffer/apm-sdk-plugin/nats-client-plugin/src/main/java/org/apache/skywalking/apm/plugin/nats/client/NatsClientPluginConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.nats.client;
+
+import org.apache.skywalking.apm.agent.core.boot.PluginConfig;
+
+public class NatsClientPluginConfig {
+
+    public static class Plugin {
+        @PluginConfig(root = NatsClientPluginConfig.class)
+        public static class NatsClient {
+            /**
+             * Nats publish message asynchronously , it put message to local queue ,
+             * then write message to network by call flush method in another thread .
+             * This config term indicate whether collect complete trace .
+             * If set to true ,the plugin will trace enqueue , flush . Otherwise, only enqueue
+             * Notice , If set true . will generate a lot of Span (one span for a message). These spans are not released util call flush
+             */
```

Comment:
According to this description, what would happen if there is `ENABLE_FULL_TRACE=false`? Is the trace going to break due to across threads between enqueue and call flush?
If so, I don't think this could be `false`. We should always trace them.

Reply:
I guess you were concerned about the payload, but that is the cost of tracing RPCs. 
Also, if this enqueue span would only be flushed later, how about using async span mode, rather than a new segment? Async span finishing costs less than a new segment. We use `capture/continues` just because there would be a complex segment including multiple spans in a new thread.

Reply:
A user maybe forget  call flush ,  or some other reason . The span will never be send ,It's main reason .
And if set true will generate a lot of span  , I think It's in  thousands .  May take up a lot of memory.
If set false , A span is released when after enqueue , But trace  not detailed enough.
So I provide this setting

Reply:
> If set false , A span is released when after enqueue , But trace not detailed enough.

Do you mean it would not have the correct RPC time, only includes a local span to show this operation? This seems strange.

Reply:
> A user maybe forget call flush , or some other reason . The span will never be send ,It's main reason .

If you want to keep tracing even when users forget to flush the data, a new segment should be good enough. Nothing would cost memory, a snapshot only includes several strings.

Reply:
If  set false , will  trace enqueue , create an exit span  for enqueue  too. 
If set true  , will trace enqueue  and  flush . A local  span for enqueue , an exit span for flush

Reply:
> If set false , will trace enqueue , create an exit span for enqueue too.

This could easily make users blaming us as bugs in two ways.
- Like you mentioned, this request may not be flushed, then SkyWalking reports an inexisting RPC.
- If this has been flushed, SkyWalking reports a wrong duration.

Both of these would be bug cases for sure. Agree?

Reply:
Agree . Maybe set default value to true is better .
If we use  a async span  to trace  enqueue and flush  (begin with enqueue , end with flush )
The RPC time is inaccurate ， Because most of time is waiting  from flush .

Reply:
> The RPC time is inaccurate ， Because most of time is waiting for flush .

Yes. 

> If we don't care about memory , I prefer two span . a local , a exit

I don't think snapshot would crash memory. The original data would cause OOM first.

Reply:
Finally , I will remove this flag . and  create two span , one localSpan for enqueue , another one exitSpan for flush .
Agree ?

Reply:
Make sense to me.